### PR TITLE
ci: Simplify PR template

### PR DESCRIPTION
**Pull request by Gudahtt**
(this text was added to test PR preview)

## **Description**

The PR template checklist has been simplified. The changes are as follows:

* The checkbox about using a draft has been replaced by a comment at the top of the template
* The checkboxes about using the template properly have been consolidated into a "I have completed the template" checkbox

This should streamline the process of creating a PR for contributors, while preserving all the same requirements.

## **Related issues**

N/A

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+<!--
+Please submit this PR as a draft initially.
+Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
```
Comment (Gudahtt) [inline .github/pull-request-template.md:3]: Not sure on exactly what criteria we should suggest here for something being ready for review, but this seemed like a good starting point

File: .github/pull-request-template.md [hunk 2]
```diff
@@ -31,16 +36,10 @@ Fixes:
 ## **Pre-merge author checklist**
 
 - [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
-- [ ] I've clearly explained what problem this PR is solving and how it is solved.
-- [ ] I've linked related issues
-- [ ] I've included manual testing steps
-- [ ] I've included screenshots/recordings if applicable
+- [ ] I've completed the PR template to the best of my ability
```
Comment (Gudahtt) [inline .github/pull-request-template.md:39]: Alternatively we could omit this, and assume the PR author has properly reviewed and filled out the template.

But having it here as an explicit step might be a useful reminder, especially in cases where a PR sits in draft for a while, with the template partially filled out. This is a good reminder to go back and review.

**Review by danjm — CHANGES_REQUESTED** [review]
This looks good to me. Were you also going to make a change to put one or more of the checkboxes in a comment?

**Comment by Gudahtt** [comment]
@danjm Sorry I am not sure what you're referring to

**Review by danjm — APPROVED** [review]
Ignore my last comment, I was misremembering something from a past conversation. This looks good to ship

File: .github/pull-request-template.md (hunk 1 above)
Comment (gauthierpetetin) [reply .github/pull-request-template.md:3]: I think the initial purpose for placing the **Description** section at the very top was to have the description previewed when PR is posted on Slack.
cc @NicolasMassart as it was initially brought up by you I believe

Comment (Gudahtt) [reply .github/pull-request-template.md:3]: Interesting! I think this comment would be omitted from the preview, but that's a great point, we should test this to make sure

Comment (Gudahtt) [reply .github/pull-request-template.md:3]: I have just tested this by adding a comment to the description of this PR, and sharing it with myself. I also added the text "(this text was added to test PR preview)" to the description to ensure I didn't get a cached copy.

It looks like only rendered content is included in the preview, so this comment will not affect it.

Comment (gauthierpetetin) [reply .github/pull-request-template.md:3]: Ok great, thanks for the test

File: .github/pull-request-template.md (hunk 2 above)
Comment (sethkfman) [reply .github/pull-request-template.md:39]: Let's leave it for now and we can refine again in a few weeks.

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by gauthierpetetin — APPROVED** [review]

**Pull request merged**